src/stream.rs

```diff
@@ -417,326 +422,357 @@ mod tests {
 
     #[test]
     fn empty_read() {
-        let mut buf = RecvBuf::default();
-        assert_eq!(buf.len(), 0);
+        let mut recv = RecvBuf::default();
+        assert_eq!(recv.len(), 0);
+
+        let mut buf = vec![0; 32];
 
-        let read = buf.pop(std::usize::MAX).unwrap();
-        assert_eq!(read.len(), 0);
-        assert_eq!(read.fin(), false);
+        let (len, fin) = recv.pop(buf.as_mut_slice()).unwrap_or((0, false));
```

Comment:
To check that `Done` is returned you can also just do:
```rust
assert_eq!(recv.pop(buf.as_mut_slice()), Err(Error::Done));
```
Same below in a few other tests.